### ISSUE TITLE
[BUG] Syntax Error:(/config/bigip.conf at line: 7346) ";" unknown property

### ISSUE THREAD

**Issue by rjouhann** [f5devcentral/f5-journeys#66]
**Describe the bug**
Opening this bug on behalf of Chris. Tried in his lab to migrate a BIG-IP from vCMP to VELOS but failed restoring the modified UCS with the following error message when trying to do a tmsh load sys config after platform-migrate command completed.

Syntax Error:(/config/bigip.conf at line: 7346) ";" unknown property

We also noticed the deployment never ends fails or succeeds in Journeys.

**To Reproduce**
1. Get UCS from SR 1-6993580048\INCOMING\jun_10\lab02.ucs
2. Restore the UCS on a vCMP guest 14.1.4.6
3. Use Journeys 3.3.2 to migrate the config to a VELOS tenant 14.1.4.6 after resolving all the errors in the tool

**Error**
[root@bipmn013lab02:/S2-red-P::Offline:Disconnected (Sync Only)] config # tmsh load sys config
Loading system configuration...
  /defaults/asm_base.conf
  /defaults/config_base.conf
  /defaults/ipfix_ie_base.conf
  /defaults/ipfix_ie_f5base.conf
  /defaults/low_profile_base.conf
  /defaults/low_security_base.conf
  /defaults/policy_base.conf
  /defaults/wam_base.conf
  /defaults/analytics_base.conf
  /defaults/apm_base.conf
  /defaults/apm_oauth_base.conf
  /defaults/apm_saml_base.conf
  /defaults/app_template_base.conf
  /defaults/classification_base.conf
2022 Jun 10 18:55:57 slot3 logger[4145]: Re-starting lind
  /var/libdata/dpi/conf/classification_update.conf
  /defaults/ips_base.conf
2022 Jun 10 18:55:58 slot1 logger[2881]: Re-starting lind
  /var/libdata/ips/ips_update.conf
  /defaults/daemon.conf
  /defaults/pem_base.conf
  /defaults/profile_base.conf
  /defaults/sandbox_base.conf
  /defaults/security_base.conf
2022 Jun 10 18:55:58 slot3 logger[4413]: Re-starting mcpd
  /defaults/urldb_base.conf
  /usr/share/monitors/base_monitors.conf
  /defaults/cipher.conf
  /defaults/ilx_base.conf
Loading configuration...
  /config/bigip_base.conf
  /config/bigip_user.conf
  /config/bigip.conf
Syntax Error:(/config/bigip.conf at line: 7346) ";" unknown property

**Context**
- Please specify, app version: 3.3.2
- And Platform version of the system you want to migrate: vCMP guest to VELOS tenant 14.1.4.6

**Comment by darkrain42** [comment]
I encountered this recently, as well.  In my original UCS archive, this was the config stanza being processed:

```
security dos bot-signature "/Common/Citrix ICA Client" {
    category "/Common/Service Agent"
    risk na
    rule "headercontent:\"\\Citrix\\ICA Client\\\"; useragentonly; nocase;"
    user-defined false
}
```

This is what Journeys generated.  The line that reports a syntax error about is the rule line.  Note that Journeys has malformed the escaping:
```
security bot-defense signature "/Common/Citrix ICA Client" {
    category "/Common/Service Agent"
    risk na
    rule headercontent:"\\Citrix\\ICA Client\\"; "useragentonly;" "nocase;"
    user-defined false
}
```

The failure is evident just trying to load that config stanza onto a running v15.1.5.1 system:

```
# cat /var/tmp/journeys-66
security bot-defense signature "/Common/Citrix ICA Client" {
    category "/Common/Service Agent"
    risk na
    rule headercontent:"\\Citrix\\ICA Client\\"; "useragentonly;" "nocase;"
    user-defined false
}
# tmsh load sys config merge verify file /var/tmp/journeys-66
Validating configuration...
  /var/tmp/journeys-66
Syntax Error:(/var/tmp/journeys-66 at line: 4) ";" unknown property
#
```

**Comment by danschuster11** [comment]
Is this a v14 problem?  The same stanza in v15 has a 'factory-rule' defined that is encoded:

security bot-defense signature "/Common/Citrix ICA Client" {
    category "/Common/Service Agent"
    factory-rule $M$wA$pbtF+8Up02HOaZ7awz6+CmNmu8+dnR0Em5qvNceInZGW5FqQ56z0EWq/AwotbC/W6e5bbSUJ5hba004YgrkR4A==
    risk na
    user-defined false
}

**Comment by darkrain42** [comment]
> Is this a v14 problem? The same stanza in v15 has a 'factory-rule' defined that is encoded:

Sort-of.

It may only occur when the source configuration file originated in v14.1.x or earlier.

In my case, Journeys was migrating a configuration from v15.1.x to v15.1.x.  I no longer have the UCS archive around, but I suspect the source device was originally on v14.1.x and was upgraded to v15.1.x at some point in the past.

In other words: it's entirely possible to have systems on v15.1.x or above that still have that particular stanza in bigip.conf.

I haven't tested, but I assume it's also possible for a user to configure a similar signature rule that trips over the incorrect escaping.

**Comment by wiksla** [comment]
The bug was fixed in release 3.3.5
